ibis/impala/tests/test_client.py

```diff
@@ -25,6 +26,18 @@ def db(con, test_data_db):
     return con.database(test_data_db)
 
 
+def test_kerberos_deps_installed(env, test_data_db):
+    # See: https://github.com/ibis-project/ibis/issues/2342
+    with pytest.raises((AttributeError, TTransportException)):
```

Comment:
I was having a look at this, and I'm not sure I understand...

This test if failing if the dependencies are missing, and passing if the dependencies are installed, correct? I see the CI green, so the second case must be true. And I think you tested that it fails without the dependencies before.

But what I'm missing is why `connect` is raising when the dependencies are installed? It's because we don't have a kerberos server in the CI?

Do you mind adding a `match=...` to `pytest.raises`, with the error message this returns? I think this will make the test more reliable, and also would make the test clearer. Also, a short comment would be helpful, with something like `connect raises a ImportError when dependencies are missing, testing that it fails with AttributeError/TTransportException because there is no kerberos available` or something like that.

Reply:
Yes, that's exactly right. Adding our own kerberos server seems fairly impractical, so instead I identified what exceptions would be raised in the following situations when there is no HDFS connection added (The HDFS connection uses a different package for kerberos auth support and throws an exception prior to the Impyla package):

1. A kerberos server exists and is correctly specified, but the dependency is not installed.
2. A kerberos server exists and is incorrectly specified, but the dependency is not installed. 
3. A kerberos server does not exist and is incorrectly specified and the dependency is installed. 
4. A kerberos server exists and is incorrectly specified and the dependency is installed.
5. A kerberos server exists, is correctly specified, and the dependency is installed. (No exception raised here)

This test should work for all of the situations above except 5. If that becomes the reality, then this becomes much easier to test anyways. For our purposes though, if the test fails it means the dependencies are missing, because those imports happen relatively early in the call stack. If we see an `AttributeError` or `TTransportException` then that implies that we got through the import statements and the exceptions we see after that should be fine, because they're caused by the lack of a kerberos server.

Reply:
I'm happy to add an additional comment, but I'm not sure how useful the match would be. The following are two examples of the errors we see in the pipeline, which are pretty generic and don't provide a lot of helpful info. I feel like directing someone to this PR discussion would be more helpful overall, but if you want the match I'm happy to add it. 

`thrift.transport.TTransport.TTransportException: Could not start SASL: b'Error in sasl_client_start (-1) SASL(-1): generic failure: GSSAPI Error: Unspecified GSS failure.  Minor code may provide more information (No Kerberos credentials available (default cache: FILE:/tmp/krb5cc_0))'`

`AttributeError: 'TSocket' object has no attribute 'isOpen'`